packages/chaire-lib-backend/src/config/server.config.ts

```diff
@@ -191,6 +353,35 @@ export const setProjectConfiguration = (newConfig: Partial<ProjectConfiguration<
             }
         });
     }
+
+    // If osrmRouting is configured using the old Preferences format and not the new one in server.config, set it to the new format
+    // FIXME Remove once the old osrmRouting format is fully deprecated
+    const osrmModesInDeprecatedFormat = Preferences.get('osrmRouting').modes;
```

Comment:
I think there might be a circular dependency as the preferences need the config to be initialized to get the default preferences from the config. I used my old config file and I don't get the warning and the process does not start on the right port.

I think here you need to get the defaultPreferences from the newConfig directly: `newConfig.defaultPreferences!.osrmRouting`.